### PR TITLE
libc: Add missing asserts for malloc

### PR DESCRIPTION
**Pull request by maska989**
JIRA: CI-386

## Description
Fixes: phoenix-rtos/phoenix-rtos-project#920

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: host-generic-pc / ia32-generic-qemu.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libc/string/string_tokpbrkstr.c [hunk 1]
```diff
@@ -886,7 +898,7 @@ TEST(string_pbrk, strpbrk_order)
 
 	testStr = testdata_createCharStr(ASCII_LENGTH);
 
-	TEST_ASSERT_NOT_NULL(reversStr);
+	TEST_ASSERT_NOT_NULL(testStr);
```
Comment (damianloew) [inline libc/string/string_tokpbrkstr.c:901]: It also has to be done in line ~770.

Comment (damianloew) [reply libc/string/string_tokpbrkstr.c:901]: And in `string_lenchr.c`

File: libc/string/string_tokpbrkstr.c [hunk 2]
```diff
@@ -729,9 +738,10 @@ TEST(string_str, empty_args)
 {
 	const char str[] = "Lorem";
 
-
 	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
 
+	TEST_ASSERT_NOT_NULL(asciiStr);
```
Comment (damianloew) [inline libc/string/string_tokpbrkstr.c:743]: Thanks for adding also those missing checks, Could you please also do it next to all `testdata_createCharStr` calls?

File: libc/stdio/stdio_file.c [hunk 3]
```diff
@@ -374,11 +374,11 @@ TEST(stdio_getput, getsputs_basic)
 
 IGNORE_TEST(stdio_getput, getsputs_readonly)
 {
-	/* 
-	Upon successful completion, fputc() shall return the value it has written. 
-	Otherwise, it shall return EOF, the error indicator for the stream shall be set 
-	and errno shall be set to indicate the error. 
-
```
Comment (damianloew) [inline libc/stdio/stdio_file.c:381]: Thanks for those fixes, however it should be done in the separate commit as it's not related to malloc asserts, maybe: `libc: remove unnecessary whitespaces`

**Review by damianloew — CHANGES_REQUESTED** [review]
When your commit fixes some issue please always leave such line in the commit description:

Fixes: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/issue_nr

**Review by damianloew — APPROVED** [review]
Thanks for fixes :)

**Review by damianloew — CHANGES_REQUESTED** [review]
EDIT: missing:
Screenshot from 2023-11-14 17-15-44

**Comment by nalajcie** [comment]
> EDIT: missing: Screenshot from 2023-11-14 17-15-44

better to use short notation (`Fixes: phoenix-rtos/phoenix-rtos-project#920`), hopefully better with external tools/ possible organization name changes, etc. 

This syntax is also required by GH docs: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

File: libc/string/string_cmp.c [hunk 4]
```diff
@@ -449,6 +467,8 @@ TEST(string_strcoll, big)
 
 	memcpy(hugeStr2, hugeStr, BIG_SIZE);
 
+	TEST_ASSERT_NOT_NULL(hugeStr);
+
```
Comment (mateusz-bloch) [inline libc/string/string_cmp.c:471]: checking hugeStr before memcpy

**Review by damianloew — APPROVED** [review]
LGTM

**Pull request merged**